### DISCUSSION TITLE
How to label issues?

### DISCUSSION THREAD

**Discussion by sstevens2** [carpentries-incubator/docker-introduction#84] [General]
In the March 2021 Maintainer meeting we said we wanted to come up with a scheme for how to use labels.  I think we should discuss labels here.

**Comment by sstevens2** [comment]
I went through all the issues and used labels as follows.  Let me know what you think.

**Types:**
- `type:enhancement` = any issue that was more of a suggestion I thought would be nice but not required for the lesson
- `type:bug1 = any issue that found something broken in the current lesson
- `type:discussion` = any issue I thought was more of a prompt for discussion (maybe we should move these to discussions? we might want to try it out here first)
- `type:clarification` = where text changes were suggested that would make the lesson more clear

**Status:**
- `status:wait` = where we are waiting on another issue or the new template
- `status:in progress` = where someone was assigned to work on it or said they were working on it

Other:
- `highpriority` = something I thought probably should needed to be fixed ASAP
- `help wanted` = where it seemed like we should make the change suggested or fix the bug and we could use a pull request
- `good first issue` = a help wanted that was relatively simple and well described

**Comment by hasenalbanna** [comment]
I think the proposed labeling scheme is a good start because it separates issue type from issue status, making filtering much easier.

One suggestion would be to make the labels more consistent by using prefixes for all categories, for example:

*** type:enhancement**
* type:bug
* type:discussion
* type:clarification

and

* status:waiting
* status:in-progress
* status:completed

For priority, a similar approach could be used:

* priority:high
* priority:medium
* priority:low

This would make labels easier to search and maintain as the project grows. I also agree that discussion-oriented issues could potentially be moved to GitHub Discussions, which would help keep the issue tracker focused on actionable work items.

**Comment by ofc-its-phyla** [comment]
> In the March 2021 Maintainer meeting we said we wanted to come up with a scheme for how to use labels. I think we should discuss labels here.


During a Maintainer meeting in March 2021, the maintainers agreed that they needed a clear system for using labels in the repository. This discussion thread was created so people could discuss and decide how labels should be used.
